File: 2022/day1.py
from util import readfile


def part1(filepath: str) -> int:
    lines = readfile(filepath, True)
    cnt = 0
    max_cnt = 0
    for l in lines:
        if l == "":
            max_cnt = max(max_cnt, cnt)
            cnt = 0
        else:
            cnt += int(l)
    return max_cnt
    
    
def part2(filepath: str, nb: int) -> int:
    lines = readfile(filepath, True)
    cnt = 0
    elves = []
    for l in lines:
        if l == "":
            elves.append(cnt)
            cnt = 0
        else:
            cnt += int(l)
    
    total = 0   
    for _ in range(nb):
        total += elves.pop(elves.index(max(elves)))
    return total


if __name__ == "__main__":
    print(part1("inputs/day1.in"))
    print(part2("inputs/day1.in", 3))

File: 2021/src/main/java/org/aoc2021/puzzles/Day1.java
package org.aoc2021.puzzles;

import java.util.List;
import java.util.stream.Collectors;

import static org.aoc2021.util.Util.readFile;

public class Day1 implements Puzzle {

    private final List<Integer> values;

    public Day1() {
        this("inputs/day1.in");
    }

    public Day1(String filepath) {
        values = readFile(filepath).stream().map(Integer::parseInt).collect(Collectors.toList());
    }

    @Override
    public String part1() {
        int result = 0;
        int old_val = Integer.MAX_VALUE;
        for (int val: values) {
            if (val > old_val)
                result += 1;
            old_val = val;
        }
        return String.valueOf(result);
    }

    @Override
    public String part2() {
        int result = 0;
        int old_val = Integer.MAX_VALUE;
        for (int i = 0; i < values.size() - 2; i++) {
            int val = sumArrayElements(i);
            if (val > old_val)
                result += 1;
            old_val = val;
        }
        return String.valueOf(result);
    }

    private int sumArrayElements(int startIndex) {
        return values.get(startIndex) + values.get(startIndex + 1) + values.get(startIndex + 2);
    }
}
